pkg/utils/utils.go

```diff
@@ -234,3 +234,46 @@ func SetAvailableResourcesForApi(groupVersion string, resources *metav1.APIResou
 
 	gvResourcesCache[groupVersion] = resources
 }
+
+func GetEnvVarValue(envVars []v1.EnvVar, key string) (string, bool) {
+	for _, envVar := range envVars {
+		if envVar.Name == key {
+			return envVar.Value, true // if key exist, return value, true
+		}
+	}
+	return "", false // if key does not exist, return "", false
+}
+
+// IsUnknownGpuResourceType check if the provided gpu resource is unknown one
+func IsUnknownGpuResourceType(resources v1.ResourceRequirements) bool {
+	basicResourceTypes := map[v1.ResourceName]struct{}{
+		v1.ResourceCPU:              {},
+		v1.ResourceMemory:           {},
+		v1.ResourceStorage:          {},
+		v1.ResourceEphemeralStorage: {},
+	}
+
+	possibleGPUResourceType := map[v1.ResourceName]struct{}{}
+
+	// Helper function to add non-basic resources from the provided ResourceList
+	addNonBasicResources := func(resources v1.ResourceList) {
+		for resourceType := range resources {
+			if _, exists := basicResourceTypes[resourceType]; !exists {
+				possibleGPUResourceType[resourceType] = struct{}{}
+			}
+		}
+	}
+
+	// Add non-basic resources from both Limits and Requests
+	addNonBasicResources(resources.Limits)
+	addNonBasicResources(resources.Requests)
+
+	// Validate GPU resource types
+	for _, gpuType := range constants.GPUResourceTypeList {
+		allowedGPUResourceName := v1.ResourceName(gpuType)
+		delete(possibleGPUResourceType, allowedGPUResourceName) // Remove allowed GPU resource if exists
+	}
+
+	// Return true if there are unknown GPU resources
+	return len(possibleGPUResourceType) > 0
```

Comment:
The purpose here was to adjust the `tensorParallelSize` value based on the GPU value within the resources, but identifying the exact GPU resource name has proven challenging. Since each vendor provides their own unique resource name rather than using a Kubernetes-native object, the options for pinpointing specific GPU resources are limited.

Currently, I see two possible approaches:

- **Using the vendor name as a basis**: This would involve recognizing and categorizing resources by vendor, though it may limit flexibility when new vendor-specific GPU resources are introduced.
- **Assuming any non-Kubernetes-native resource as GPU**: Here, we would classify anything not recognized as a standard Kubernetes resource (CPU, memory, etc.) as a GPU. This could potentially simplify detection but would require careful management to avoid misclassifications.
Do you have any suggestions for more dynamic solutions to identify GPU resources effectively?

Reply:
I added a new annotation to allow customGPUResourceTypes in isvc.
(ex)
```
serving.kserve.io/gpu-resource-types: '["gpu-type1", "gpu-type2", "gpu-type3"]'
```
if users want to specific gpu type, they can set this annotation.

Basic behavior:
- only allows 4 types 
  ~~~
  	NvidiaGPUResourceType = "nvidia.com/gpu"
	AmdGPUResourceType    = "amd.com/gpu"
	IntelGPUResourceType  = "intel.com/gpu"
	GaudiGPUResourceType  = "habana.ai/gaudi"
  ~~~
- adding additional custom gpu types to allowable gpu types
  Users can add additional custom GPU types by using the aforementioned annotation. These custom types must be specified in JSON format within the annotation value. (added validation check for this)